### PR TITLE
lint and prettier for angular app

### PR DESCRIPTION
**Pull request by pflopez**
PLAN-342

Added lint and prettier for format and glory.
We do have some things to fix, but instead of jumping the gun on some "fixes" that might have consequences, I've marked some of the rules as warnings instead of errors.

Ideally we should set up our editors to lint on save. That way we'll get the same format across the board and we'll get computers to do format for us.
Here is a guide on how to do it on VScode or Webstorm:
https://itnext.io/configure-prettier-and-eslint-with-angular-e7b4ce979cd8

You can also do it manually by running 
```
npm run lint
```
To see if you have some issues or
```
npm run lint:fix
```
To fix lint and format automatically

This pr has A LOT OF CHANGES given that its applying the linting rules. The rules can be tweaked if we want some changes. The only two non-default options I've added are [trailing commas](https://prettier.io/docs/en/options.html#trailing-commas), [braket line](https://prettier.io/docs/en/options.html#trailing-commas)  and [single quotes](https://prettier.io/docs/en/options.html#quotes)

See `.pretttierrc` for format configuration and `.eslintrc.json` for es lint rules

### CODE REVIEW

File: src/interface/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        "createDefaultProgram": true
+      },
+      "extends": [
+        "plugin:@angular-eslint/recommended",
+        "plugin:@angular-eslint/template/process-inline-templates",
+        "plugin:prettier/recommended"
+      ],
+      "rules": {
+        "@angular-eslint/directive-selector": [
+          "warn",
```
Comment (pflopez) [inline src/interface/.eslintrc.json:24]: We have some components and directives that do not use the `app` prefix. We might want to fix that eventually but not throw an error, at least for now

File: src/interface/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    },
+    {
+      "files": [
+        "*.html"
+      ],
+      "extends": [
+        "plugin:@angular-eslint/template/recommended"
+      ],
+      "rules": {
+        "@angular-eslint/template/no-negated-async": ["warn"]
```
Comment (pflopez) [inline src/interface/.eslintrc.json:49]: Another rule that throws a lot of errors is negating asyncs, which we are doing in a lot of places. Changing this on the code might have some side effects so marking this as warning for now

File: src/interface/src/app/top-bar/top-bar.component.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ import { Region, RegionOption, regionOptions } from '../types';
 import { AccountDialogComponent } from './../account-dialog/account-dialog.component';
 
 @Component({
-  selector: 'top-bar',
+  selector: 'app-top-bar',
```
Comment (pflopez) [inline src/interface/src/app/top-bar/top-bar.component.ts:17]: I've updated this selector as it only had one usage

File: src/interface/src/app/shared/opacity-slider/opacity-slider.component.ts [hunk 4]
```diff
@@ -1,31 +1,33 @@
...
 })
-export class OpacitySliderComponent implements OnInit, OnChanges {
-  @Input() opacity: number | null | undefined = FrontendConstants.MAP_DATA_LAYER_OPACITY;
-  @Input() label: string | null = "";
-  @Output() change = new EventEmitter<number>();
+export class OpacitySliderComponent implements OnChanges {
+  @Input() opacity: number | null | undefined =
+    FrontendConstants.MAP_DATA_LAYER_OPACITY;
+  @Input() label: string | null = '';
+  @Output() changeOpacity = new EventEmitter<number>();
```
Comment (pflopez) [inline src/interface/src/app/shared/opacity-slider/opacity-slider.component.ts:20]: Renamed `change` for `changeOpacity` as it was throwing an es error

**Review by runforyourlives — APPROVED** [review]
Thank you for doing this.

It feels like it is time to add a Style section to the wiki - can you follow up with that?  We can add instructions for keeping our python code consistent as well.

**Pull request merged**